### ISSUE TITLE
Enhance echo cancellation seetings UI

### ISSUE THREAD

**Issue by Krzmbrzl** [mumble-voip/mumble#4110]
**Context**
AudioInput settings page -> echo cancellation (see #4108 for more details)

**Describe the feature you have in mind**
The option for echo cancellation should

1. Show a tooltip why it currently is disabled (if it is disabled that is)
2. update its enabled-state according to the settings that are put into the settings page right now rather than according to what the settings were when the widget was created

**Comment by Krzmbrzl** [comment]
So the problem is that whether the echo cancellation combo box is set to be  enabled, happens on a selection change of the audio input interface:
https://github.com/mumble-voip/mumble/blob/d945acb883186135143121ee2a9fcc501f3c52f7/src/mumble/AudioConfigDialog.cpp#L378
The value it uses for checking however is the one that is currently stored in the settings object the ConfigDialog is working on (so its local copy if you will). This means that the changes made in the AudioOutput page, won't be reflected here until the user clicks apply. As the enable-state is only changed on a change of the input interface though the user has to additionally change the input interface once more in order for the enable-logic to kick in and work properly again.

**Comment by toby63** [comment]
@Krzmbrzl 
An additional idea i had when reading this:
_Its low priority of course._

**tldr:** check if audio filters (like echo cancel) are already in use (e.g. in pulseaudio).

I don't know if it is a problem for mumble (I did not test yet), but in other applications (e.g. webrtc in browsers) I noticed a problem, when echo cancelling etc. is already enabled in some other form (for example in pulseaudio).
The effects can then (in my impression) distort/disturb each other.
So my request/idea is, whether its possible for mumble to notice/check if this is the case and show that in the UI as well and maybe even show a notification to the user.

More information:
Of course mumble might not be able to check for every possible option where echo cancel (or similar things) is/are already enabled somewhere else, but at least it could maybe check for some, like:
* pulseaudio
* pulseeffects
Technical hint: It might be already visible in the names of sink and source, so that mumble could check that.

#Alternatives: 
On the other hand, for users like me who use this kind of audio filters, it should be clear to deactivate them in mumble or check whether mumble is using its own audio filters.

**Comment by toby63** [comment]
Also one smaller thing:
I noticed that it is not explained what the difference is between:
* mixed echo cancel
* multichannel echo cancel 

So maybe a tooltip with a little explanation could be added, to help users decide which one to choose.

**Comment by Krzmbrzl** [comment]
> check if audio filters (like echo cancel) are already in use (e.g. in pulseaudio).

Sounds like a good idea, but I don't really know how this would be done (for figuring out whether it's enabled it should be doable (at least on Linux) but on Mumble's side I don't know where this whole stuff is done) ^^

> I noticed that it is not explained what the difference is between:
mixed echo cancel
multichannel echo cancel

You're right. I myself have no clue though so I can't add it in. If someone knows the difference, please let me know and I'll add it to the tooltip :)

**Comment by streaps** [comment]
what lib is used for the echo cancelling?

**Comment by streaps** [comment]
> #Alternatives:
> On the other hand, for users like me who use this kind of audio filters, it should be clear to deactivate them in mumble or check whether mumble is using its own audio filters.

+1
I don't think it's Mumble's responsibility to check for active Pulseaudio modules that are not enabled by default. Teamspeak, Firefox, Chrome, Skype don't do anything about it.

I also don't think echo cancellation belongs in the audio server (but that is debatable). I would prefer to have better echo cancellation in Mumble, the one from Speex is a bit outdated.

It's also not unlikely that pulseaudio will be replaced by pipewire at some point.

**Comment by streaps** [comment]
> I noticed that it is not explained what the difference is between:
>     * mixed echo cancel
>     * multichannel echo cancel

Is multichannel echo cancel useful at all?

**Comment by Krzmbrzl** [comment]
No idea :shrug:

**Comment by toby63** [comment]
@streaps 
> I don't think it's Mumble's responsibility to check for active Pulseaudio modules that are not enabled by default. Teamspeak, Firefox, Chrome, Skype don't do anything about it.

Well, it would only be for convenience, like many other things are too.

> I also don't think echo cancellation belongs in the audio server (but that is debatable). I would prefer to have better echo cancellation in Mumble, the one from Speex is a bit outdated.

Thats exactly the point, I see e.g. that the webrtc method in the pulse module for echo-cancel is much better than what is used in many programs.
By the way. no offense to the mumble team, these things are of course very complicated. 
But because of ~~AI~~ I don't like this term, I would call it "limited selflearning algorithms" - there is hope on the horizon.
And I already saw that rnnoise seems to be included here, but it is not as far advanced.
(yeah i know, noice cancelation is not exactly the same as echo cancel)

**Comment by toby63** [comment]
@Krzmbrzl 
Should we/I open new issues for the things I mentioned ([1](https://github.com/mumble-voip/mumble/issues/4110#issuecomment-621297205), [2](https://github.com/mumble-voip/mumble/issues/4110#issuecomment-621290525))?

**Comment by Krzmbrzl** [comment]
yeah :)

**Comment by toby63** [comment]
ok

**Comment by toby63** [comment]
Opened #4125 and #4127.
